avr/variants/sanguino/pins_arduino.h

```diff
@@ -234,8 +241,13 @@ const uint8_t PROGMEM digital_pin_to_timer_PGM[] =
         TIMER0A,        /* 3  - PB3 */
         TIMER0B,        /* 4  - PB4 */
         NOT_ON_TIMER,   /* 5  - PB5 */
+#if  defined(__AVR_ATmega644__) || defined(__AVR_ATmega644P__)
         NOT_ON_TIMER,   /* 6  - PB6 */
         NOT_ON_TIMER,   /* 7  - PB7 */
+#elif defined(__AVR_ATmega1284__) || defined(__AVR_ATmega1284P__)
+        TIMER3A,        /* 6  - PB6 */
+        TIMER3B,        /* 7  - PB7 */ 
+#endif        
```

Comment:
I think it would be better to replace this with:
```C++
#ifdef TCCR3A
    TIMER3A,        /* 6  - PB6 */
    TIMER3B,        /* 7  - PB7 */ 
#else
    NOT_ON_TIMER,        /* 6  - PB6 */
    NOT_ON_TIMER,        /* 7  - PB7 */ 
#endif
```